src/main/kotlin/no/nav/syfo/behandlendeenhet/EnhetService.kt

```diff
@@ -57,11 +57,14 @@ class EnhetService(
                 personIdentNumber = personIdent,
                 veilederToken = veilederToken,
             )
-            val newBehandlendeEnhet = if (enhetId?.value != geografiskEnhet.enhetId) enhetId else null
+            val newBehandlendeEnhet = if (enhetId != geografiskEnhet.enhetId) enhetId else null
             val currentOppfolgingsenhet = getOppfolgingsenhet(personIdent)
             val navIdent = veilederToken?.getNAVIdent() ?: SYSTEM_USER_IDENT
             if (newBehandlendeEnhet != null || currentOppfolgingsenhet != null) {
-                repository.createOppfolgingsenhet(personIdent, newBehandlendeEnhet, navIdent).also {
+                val pOppfolgingsenhet = repository.createOppfolgingsenhet(personIdent, newBehandlendeEnhet, navIdent)
+                pOppfolgingsenhet.toOppfolgingsenhet(
+                    enhetNavn = newBehandlendeEnhet?.let { getEnhetsnavn(it) },
+                ).also {
                     behandlendeEnhetProducer.sendBehandlendeEnhetUpdate(it, it.createdAt)
                 }
             } else {
```

Comment:
Ønsker vi å gjøre noe eller logge noe i `else`'n her, eller det fint at den stille ikke foretar seg noe 🤔 Er det i tilfeller hvor vedkommende ikke har oppfølgingsenhet, og `enhetId` som kommer inn i funksjonen heller ikke er noe annet enn den geografiske?

Reply:
Ja, var her vi logga litt før. Den kommer inn her når cronjobben i `syfooversiktsrv` kjøres 1 gang i døgnet for å sjekke om det er noen personer som ikke har vært sykmeldt på 2 mnd skal "nullstille" oppfølgingsenhet og tildelt veileder. Da sender man inn `enhetId = null`, og de finnes heller nødvendigvis ikke i databasen her (fordi de aldri har blitt tildelt noen annen enhet), så da blir `currentOppfolgingsenhet` også `null`.

Reply:
Ja, de case'ene vi kommer hit er ikke indikasjon på at noe var feil, men bare det at det ikke var noe som trengtes å gjøres.